llama-index-integrations/llms/llama-index-llms-vertex/llama_index/llms/vertex/base.py

```diff
@@ -16,6 +16,7 @@
 from llama_index.core.llms.callbacks import llm_chat_callback, llm_completion_callback
 from llama_index.core.llms.llm import LLM
 from llama_index.core.types import BaseOutputParser, PydanticProgramMode
+from llama_index.llms.gemini.utils import merge_neighboring_same_role_messages
```

Comment:
This places a dependency on Gemini from Vertex - which prolly isn't ideal.
An alternative would be to copy this function over to `llama-index-integrations/llms/llama-index-llms-vertex/llama_index/llms/vertex/gemini_utils.py`

But an even better solution would be to move it to some common folder that both Gemini and Vertex can import from. Where is that folder?

Reply:
probably this function could be moved to core, in some LLM utils file?